bindings/wasm/src/credential/credential_builder.rs

```diff
@@ -0,0 +1,161 @@
+    }
+
+    Ok(builder)
+  }
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "ICredential")]
+  pub type ICredential;
+}
+
+/// Fields for constructing a new {@link Credential}.
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[typescript(name = "ICredential", readonly, optional)]
+struct ICredentialHelper {
+  /// The JSON-LD context(s) applicable to the `Credential`.
+  #[typescript(type = "string | Record<string, any> | Array<string | Record<string, any>>")]
+  context: Option<OneOrMany<Context>>,
+  /// A unique URI referencing the subject of the `Credential`.
+  #[typescript(type = "string")]
+  id: Option<String>,
+  /// One or more URIs defining the type of the `Credential`. Contains the base context by default.
+  #[typescript(name = "type", type = "string | Array<string>")]
+  r#type: Option<OneOrMany<String>>,
+  /// One or more objects representing the `Credential` subject(s).
+  #[typescript(optional = false, name = "credentialSubject", type = "Subject | Array<Subject>")]
+  credential_subject: Option<OneOrMany<Subject>>,
+  /// A reference to the issuer of the `Credential`.
+  #[typescript(optional = false, type = "string | DID | Issuer")]
+  issuer: Option<Issuer>,
+  /// A timestamp of when the `Credential` becomes valid. Defaults to the current datetime.
+  #[typescript(name = "issuanceDate", type = "Timestamp")]
+  issuance_date: Option<Timestamp>,
+  /// A timestamp of when the `Credential` should no longer be considered valid.
+  #[typescript(name = "expirationDate", type = "Timestamp")]
+  expiration_date: Option<Timestamp>,
+  /// Information used to determine the current status of the `Credential`.
+  #[typescript(name = "credentialStatus", type = "Status | Array<Status>")]
+  credential_status: Option<OneOrMany<Status>>,
+  /// Information used to assist in the enforcement of a specific `Credential` structure.
+  #[typescript(name = "credentialSchema", type = "Schema | Array<Schema>")]
+  credential_schema: Option<OneOrMany<Schema>>,
+  /// Service(s) used to refresh an expired `Credential`.
+  #[typescript(name = "refreshService", type = "RefreshService | Array<RefreshService>")]
+  refresh_service: Option<OneOrMany<RefreshService>>,
+  /// Terms-of-use specified by the `Credential` issuer.
+  #[typescript(name = "termsOfUse", type = "Policy | Array<Policy>")]
+  terms_of_use: Option<OneOrMany<Policy>>,
+  /// Human-readable evidence used to support the claims within the `Credential`.
+  #[typescript(type = "Evidence | Array<Evidence>")]
+  evidence: Option<OneOrMany<Evidence>>,
+  /// Indicates that the `Credential` must only be contained within a {@link Presentation} with a proof issued from the
+  /// `Credential` subject.
+  #[typescript(name = "nonTransferable", type = "boolean")]
+  non_transferable: Option<bool>,
+  /// Miscellaneous properties.
+  #[serde(flatten)]
+  #[typescript(optional = false, name = "[properties: string | symbol]", type = "unknown")]
```

Comment:
I don't get an error setting a `Symbol` key if I remove the `readonly` property, but it doesn't show up in the logs so I think it's best to remove it, thanks.

> It also can produce undesired behavior with serde since you can set 2 keys with the same string representation

We have this problem in Rust too with the `properties` map, it's just something we have to accept can happen.